src/serialization.c

```diff
@@ -253,7 +202,15 @@ RRStatus RaftRedisCommandArrayDeserialize(RaftRedisCommandArray *target, const v
         RaftRedisCommandArrayFree(target);
     }
 
-    /* Read multibulk count */
+    /* Read asking */
+    size_t asking;
+    if ((n = decodeInteger(p, buf_size, '*', &asking)) < 0) {
+        return RR_ERROR;
+    }
+    p += n; buf_size -= n;
```

Comment:
```suggestion
    p += n; 
    buf_size -= n;
```

Reply:
this I'm less sure of, i sot of like pointer moving / buf shrinking being together.  but don't have strong opinions on it

Reply:
Oh, this was intentional. Okay, just ignore my comment, we can leave it as it is. 

It was just a bit surprising to me that two statements at the same line. just a note, I think common pattern to avoid double variable adjustment is calculating end pointer.

```c
char *p = start;
char *end = p + len;

p++;
p+=15;
p+=30;

passYourPointerWithLength(p, end - p);
```